### PR TITLE
EDSC-2683: Adds tree searching

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: src/components/Tree/Tree.js [hunk 1]
```diff
@@ -132,9 +135,21 @@ export const Tree = ({
     updateModel()
   }
 
+  /**
+   * Sets the filterText state
+   * @param {Object} e event object
+   */
+  const onFilterChange = (e) => {
+    setFilterText(e.target.value.toLowerCase())
```
Comment (abbottry) [inline src/components/Tree/Tree.js]: May want to trim this as well while you're formatting -- how leading/trailing whitespace function?

File: src/util/TreeNode.js [hunk 2]
```diff
@@ -215,6 +217,24 @@ export class TreeNode {
     return checked
   }
 
+  /**
+   * Determines if the node matches the given text
+   * @param {String} filterText text to filter the node on
+   */
+  matchesFilter(filterText) {
+    // return this.value.toLowerCase().indexOf(filterText) !== -1 || this.label.toLowerCase().indexOf(filterText) !== -1
```
Comment (abbottry) [inline src/util/TreeNode.js]: Debug?

Comment (macrouch) [reply src/util/TreeNode.js]: :doh:

File: src/components/Tree/Tree.js (hunk 1 above)
Comment (macrouch) [reply src/components/Tree/Tree.js]: I just tried trimming it here and that doesn't let you add a space at all. I added a trim when passing it to TreeItem and that behaves better

File: src/components/Tree/Tree.js [hunk 3]
```diff
@@ -149,7 +150,7 @@ export const Tree = ({
   const nodeList = () => treeModel.current.children.map(child => (
     <TreeItem
       key={`${child.elementHash}`}
-      filterText={filterText}
+      filterText={filterText.trim()}
```
Comment (abbottry) [inline src/components/Tree/Tree.js]: Seems odd/bad to run operations on props as we provide them. Can we pull it out of the assignment and do it before we construct the TreeItem or determine why its messing up where you are doing the lowercase? `.trim()` should not remove whitespace inside strings, it should be a combo of `trimStart` and `trimEnd`.

**Review by abbottry — APPROVED** [review]

**Pull request merged**